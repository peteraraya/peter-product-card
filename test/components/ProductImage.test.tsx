import * as React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from "../../src/components";
import { product2 } from '../data/products';
describe('ProductImage', () => { 

  test('debe mostrar el componente ', () => {
    const wrapper = renderer.create(
      <ProductImage img={`https://upload.wikimedia.org/wikipedia/commons/thumb/4/49/Flag_of_Ukraine.svg/2560px-Flag_of_Ukraine.svg.png`} />
    );
    expect(wrapper).toMatchSnapshot();
    // console.log(wrapper.toJSON());

  });


  test('debe mostrar el componente correctamente con la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2} >
        {() => <ProductImage />}
      </ProductCard>
    );
    expect(wrapper).toMatchSnapshot();

  });

 })