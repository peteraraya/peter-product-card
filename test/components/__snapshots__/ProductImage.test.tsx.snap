// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente  1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://upload.wikimedia.org/wikipedia/commons/thumb/4/49/Flag_of_Ukraine.svg/2560px-Flag_of_Ukraine.svg.png"
/>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
